### PR TITLE
WIP: Adding barebones tests for the proxy endpoints

### PR DESCRIPTION
**Pull request by arschles**
This is a barebones example on how to write a test for the `/@v/list` endpoint for @fedepaol to look at

Ref #138

### CODE REVIEW

File: cmd/proxy/actions/list_test.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package actions
+
+func (a *ActionSuite) TestList() {
+	r := a.Require()
+
+	req := a.JSON("/github.com/gomods/athens/@v/list")
```
Comment (marwan-at-work) [inline cmd/proxy/actions/list_test.go]: Let's have all repository examples be stemmed from `github.com/athens-artifacts`, this way we can actually test the response and be sure it won't change overtime :)

**Comment by robjloranger** [comment]
@fedepaol was this just meant as a demo for you?

**Comment by fedepaol** [comment]
@robjloranger not really. I was mentioned in the very first commit in order to take a look at it for testing the buffalo middleware, but I am pretty sure @arschles meant to have this working in order to test the handlers.

**Comment by arschles** [comment]
I'd like to leave this PR open (with `do not merge`) so that I can finish the actions tests (I had been doing this in a screencast, which I'd like to finish 😄 )

**Comment by arschles** [comment]
Ok, I'm finally closing this. I had been holding on to this romantic dream that I'd come back and write the most awesome-est tests in the world, but I haven't. And now, there are conflicts, oh no!

If I come back to write tests, I'll open a new PR. If anyone else wants to write these unit tests, we don't have an issue for that but you're welcome to!

**Pull request closed without merging**